core/src/main/java/org/alfresco/error/ExceptionStackUtil.java

```diff
@@ -38,10 +38,16 @@ public static Throwable getCause(Throwable throwable, Class<?> ... possibleCause
     {
         while (throwable != null)
         {
+            Class<?> throwableClass = throwable.getClass();
+
+            boolean isJavaScriptException = throwableClass.getName().contains("JavaScriptException");
+            String throwableMsg = throwable.getMessage() != null ? throwable.getMessage() : "";
+
             for (Class<?> possibleCauseClass : possibleCauses)
             {
-                Class<?> throwableClass = throwable.getClass();
-                if (possibleCauseClass.isAssignableFrom(throwableClass))
+                String possibleCauseClassName = possibleCauseClass.getName();
+
+                if (possibleCauseClass.isAssignableFrom(throwableClass) || (isJavaScriptException && throwableMsg.contains(possibleCauseClassName)))
```

Comment:
Is there a delimiter we could use?  Sometimes class names contain each other.

Reply:
Added a colon as delimiter.